src/common/components/cards/cards-visualization-modifier-buttons.tsx

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import {
+    cardsVisualizationModifiersContainer,
+    expandCollapseAllButton,
+    visualHelperToggle,
+} from 'common/components/cards/cards-visualization-modifiers-buttons.scss';
+import { CardSelectionMessageCreator } from 'common/message-creators/card-selection-message-creator';
+import { NamedFC } from 'common/react/named-fc';
+import { SupportedMouseEvent } from 'common/telemetry-data-factory';
+import { ActionButton, Toggle } from 'office-ui-fabric-react';
+import * as React from 'react';
+
+export type CardsVisualizationModifierButtonsDeps = {
+    cardSelectionMessageCreator: CardSelectionMessageCreator;
+};
+
+export type CardsVisualizationModifierButtonsProps = {
+    deps: CardsVisualizationModifierButtonsDeps;
+    visualHelperEnabled: boolean;
+    allCardsCollapsed: boolean;
+};
+
+export const CardsVisualizationModifierButtons = NamedFC<CardsVisualizationModifierButtonsProps>(
+    'CardsVisualizationModifierButtons',
+    props => {
+        const { deps, visualHelperEnabled, allCardsCollapsed } = props;
+
+        let onClick: (_: SupportedMouseEvent) => void = deps.cardSelectionMessageCreator.collapseAllRules;
```

Comment:
- is it necesary to wrap `cardSelectionMessageCreator.collapseAllRules` here? is it because of the event type doesn't match?
- the name here, `onClick` is not to informative, as we have to `onClick` props down below (in the `ActionButton` and the `Toggle`). I do see that the on click function changes base on the state. Still feels to generic for a name.

Reply:
Not necessary; updated.